pkg/assembler/backends/neo4j/artifact.go

```diff
@@ -45,3 +50,74 @@ func (an *artifactNode) IdentifiablePropertyNames() []string {
 	// An artifact can be uniquely identified by algorithm and digest
 	return []string{"algorithm", "digest"}
 }
+
+func (c *neo4jClient) Artifacts(ctx context.Context, artifactSpec *model.ArtifactSpec) ([]*model.Artifact, error) {
+	session := c.driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead})
+	defer session.Close()
+
+	result, err := session.ReadTransaction(
+		func(tx neo4j.Transaction) (interface{}, error) {
+
+			var sb strings.Builder
+			var firstMatch bool = true
+			queryValues := map[string]any{}
+
+			sb.WriteString("MATCH (n:Artifact)")
+			if artifactSpec.Algorithm != nil {
```

Comment:
nit: i think for artifact should we mandate both be specified or not at all?

Reply:
+1, we need both

Reply:
add new error case:
```go
	if artifactSpec.Algorithm == nil && artifactSpec.Digest == nil {
		return nil, gqlerror.Errorf("must specify both algorithm and digest for artifact")
	}
```